sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -88,12 +88,24 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       query: Dataset[_],
       tableName: Option[String] = None,
       storageLevel: StorageLevel = MEMORY_AND_DISK): Unit = {
-    val planToCache = query.logicalPlan
+    cacheQueryWithLogicalPlan(query.sparkSession, query.logicalPlan, tableName, storageLevel)
+  }
+
+  /**
+   * Caches the data produced by the given [[LogicalPlan]].
+   * Unlike `RDD.cache()`, the default storage level is set to be `MEMORY_AND_DISK` because
+   * recomputing the in-memory columnar representation of the underlying table is expensive.
+   */
+  def cacheQueryWithLogicalPlan(
```

Comment:
shall we just call it `cacheQuery` to be consistent with `uncacheQuery`?

Reply:
Unfortunately, we cannot because Scala doesn't allow overloading methods with default arguments.

Reply:
Ah, got it. Then how about `cacheLogicalPlan` to be simpler?

Reply:
Ah, I think we can do the folllowing:
```scala
  def cacheQuery(
    spark: SparkSession,
    planToCache: LogicalPlan,
    tableName: Option[String]): Unit = {
    cacheQuery(spark, planToCache, tableName, MEMORY_AND_DISK)
  }

  def cacheQuery(
      spark: SparkSession,
      planToCache: LogicalPlan,
      tableName: Option[String],
      storageLevel: StorageLevel)
```